pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -139,7 +140,9 @@ public String toString() {
 
     @Override
     public ConcurrentOpenHashSet<Producer> getProducers() {
-        return producers;
+        ConcurrentOpenHashSet<Producer> result = new ConcurrentOpenHashSet<>(16, 1);
```

Comment:
oops. I didn't realize that there is a `getProducers` here. It is used in a lot of places. hence it might be a performance problem if we changed to a hash map.

Reply:
I have check the used places of method `getProducers()`
![image](https://user-images.githubusercontent.com/12592133/68322254-11e2f080-00fe-11ea-8a27-7ea28d1e0bf2.png)
Looks ok, two places used it, handle backlog exceeded backlog and prometheus metrics aggregator, others are unit tests.

Reply:
@sijie i will try change the getProducers method to return a map.